validation/src/shared_table/mod.rs

```diff
@@ -541,6 +555,18 @@ impl SharedTable {
 	}
 }
 
+fn convert_misbehaviour(
+	key: SessionKey,
+	misbehaviour: GenericMisbehavior<CandidateReceipt, Hash, ValidatorIndex, ValidatorSignature>
+) -> table::Misbehavior {
+	match misbehaviour {
+		GenericMisbehavior::ValidityDoubleVote(val) => table::Misbehavior::ValidityDoubleVote(val),
+		GenericMisbehavior::MultipleCandidates(val) => table::Misbehavior::MultipleCandidates(val),
+		GenericMisbehavior::UnauthorizedStatement(val) => table::Misbehavior::UnauthorizedStatement(val),
+		GenericMisbehavior::DoubleSign(val) => table::Misbehavior::DoubleSign(val),
```

Comment:
This will be ugly because "enum variants on type aliases are experimental".

Reply:
Shouldn't the type alias encompass `ValidatorIndex` already?